integrationtests/tests/fixtures/mod.rs

```diff
@@ -233,7 +234,8 @@ impl GatewayTest {
             client_config: client_config.clone(),
             redeem_key: kp,
             timelock_delta: 10,
-            api: format!("http://{}", bind_addr),
+            api: Url::parse(format!("http://{}", bind_addr).as_str())
+                .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
```

Comment:
I think we decided to tell why we expected something to never fail in the expect messages.

Reply:
This is actually going to fail for v6 addresses, because in URLs the IP needs be put into `[ ... ]`. You probably want to put this conversion in a separate function and write couple of quick unit tests for it.

Reply:
good idea, thanks for the heads up @dpc